theforeman.org/pipelines/test/foreman/integration.groovy

```diff
@@ -0,0 +1,65 @@
+def ruby_versions = foreman_assets_versions[ghprbTargetBranch]['ruby']
+
+pipeline {
+    agent { label 'fast' }
+
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        ansiColor('xterm')
+    }
+
+    stages {
+        stage('matrix') {
+            matrix {
+                agent { label 'fast' }
+                axes {
+                    axis {
+                        name 'ruby'
+                        values '2.7'
```

Comment:
Would this work to always run only on the first entry (which should be the latest):
```suggestion
                        values ruby_versions[0]
```

Then the `when` part below is redundant. You can also use `foreman_unit_versions` and get rid of the duplication.

Reply:
I could try this, but what you pass to this cannot be dynamic from my previous playing around

Reply:
I expected that when we started turning on Ruby 3.0 support, we were going to want to run integration tests against 3.0 for at least some period of time to ensure things worked in both environments.

Reply:
That's a fair point.